src/test/groovy/synacy/gradprogram/spock/exercise/RefundServiceSpec.groovy

```diff
@@ -0,0 +1,131 @@
+        }
+
+        where:
+        orderCancelReason << [CancelReason.DAMAGED, CancelReason.WRONG_ITEM]
+    }
+
+    def "createAndSaveRefundRequest should save a refund request with a half refund with WRONG_ITEM CancelReason and is cancelled in after 4 days"() {
+        given:
+        def order = Mock(Order)
+        def cancelRequest = Mock(CancelOrderRequest)
+
+        def orderUUID = UUID.randomUUID()
+        def recipientName = "Pope Francis"
+        def orderTotalCost = 500.0
+        def mockDate = generateMockOrderDate()
+        def daysSinceOrder = 4
+
+        order.getId() >> orderUUID
+        order.getRecipientName() >> recipientName
+        order.getTotalCost() >> orderTotalCost
+        order.getDateOrdered() >> mockDate
+
+        cancelRequest.getReason() >> CancelReason.WRONG_ITEM
+
+        this.dateUtils.calculateDifferenceInDays(_ as Date,_ as Date) >> daysSinceOrder
+
+        when:
+        this.refundService.createAndSaveRefundRequest(order, cancelRequest)
+
+        then:
+        1 * this.refundRepository.saveRefundRequest(_ as RefundRequest) >> { RefundRequest refundRequest ->
+            assert refundRequest.getRecipientName() == recipientName
+            assert refundRequest.getOrderId() == orderUUID
+            assert refundRequest.getRefundAmount() == BigDecimal.valueOf(250.0);
+            assert refundRequest.getStatus() == RefundRequestStatus.TO_PROCESS
+        }
+    }
+
+    def "createAndSaveRefundRequest should save a refund request with full refund with DAMAGED CancelReason and is cancelled after 4 days"() {
+        given:
+        def order = Mock(Order)
+        def cancelRequest = Mock(CancelOrderRequest)
+
+        def orderUUID = UUID.randomUUID()
+        def recipientName = "Pope Francis"
+        def orderTotalCost = 500.0
+        def daysSinceOrder = 4
+
+        order.getId() >> orderUUID
+        order.getRecipientName() >> recipientName
+        order.getTotalCost() >> orderTotalCost
+
+        cancelRequest.getReason() >> CancelReason.DAMAGED
+
+        this.dateUtils.calculateDifferenceInDays(_ as Date,_ as Date) >> daysSinceOrder
+
+        when:
+        this.refundService.createAndSaveRefundRequest(order, cancelRequest)
+
+        then:
+        1 * this.refundRepository.saveRefundRequest(_ as RefundRequest) >> { RefundRequest refundRequest ->
+            assert refundRequest.getRecipientName() == recipientName
+            assert refundRequest.getOrderId() == orderUUID
+            assert refundRequest.getRefundAmount() == BigDecimal.valueOf(500.0);
+            assert refundRequest.getStatus() == RefundRequestStatus.TO_PROCESS
+        }
+    }
+
+    def generateMockOrderDate() {
+        LocalDate localDate = LocalDate.now().minusDays(RefundConstants.MAX_DAYS_FOR_FULL_REFUND + 1)
+        Date mockDate = Date.from(localDate.atStartOfDay(ZoneId.systemDefault()).toInstant());
+
+        return mockDate
+    }
```

Comment:
I don't think you need to do this since you have extracted out the logic to calculate the difference between the order date and you can just control is via stubbing of that method.